Gems/AWSCore/Code/Tests/TestFramework/AWSCoreFixture.h

```diff
@@ -120,14 +126,33 @@ class AWSCoreFixture
         AZ::IO::FileIOBase::SetInstance(nullptr);
         AZ::IO::FileIOBase::SetInstance(m_localFileIO);
 
-        m_settingsRegistry = AZStd::make_unique<AZ::SettingsRegistryImpl>();
-        AZ::SettingsRegistry::Register(m_settingsRegistry.get());
+        if (mockSettingsRegistry)
+        {
+            m_settingsRegistry = AZStd::make_unique<AZ::SettingsRegistryImpl>();
+            AZ::SettingsRegistry::Register(m_settingsRegistry.get());
+        }
+        else
+        {
+            m_app = AZStd::make_unique<AZ::ComponentApplication>();
```

Comment:
Curious: Do you need to set m_app as well when mockSettingsRegistry is true?

Reply:
no, I don't need when setting registry is mocked.